module/VuFind/src/VuFind/View/Helper/Root/ThemeConfigFactory.php

```diff
@@ -69,15 +69,16 @@ public function __invoke(
             throw new \Exception('Unexpected options sent to factory.');
         }
 
-        $themeInfo = $container->get(\VuFindTheme\ThemeInfo::class);
-
         // As of release 1.1.0, the memory storage adapter has a flaw which can cause
         // unnecessary out of memory exceptions when a memory limit is enabled; we
         // can disable these problematic checks by setting memory_limit to -1.
         $cacheConfig = ['name' => 'memory', 'options' => ['memory_limit' => -1]];
         $cache = $container->get(\Laminas\Cache\Service\StorageAdapterFactory::class)
             ->createFromArrayConfiguration($cacheConfig);
 
-        return new $requestedName($themeInfo, $cache);
+        $themeInfo = $container->get(\VuFindTheme\ThemeInfo::class);
+        $themeInfo->setCache($cache);
```

Comment:
I think it would make much more sense to set the cache in the factory for ThemeInfo rather than here. We don't want the factory for one service to modify a different service as a side effect.